### PR TITLE
feature/teaching/#45

### LINKED ISSUE #45: As a user, I can select which course I would like to manage now, so that I can access information about the TAs under my course easily.

### PR DESCRIPTION
**Pull request by itssisi**
Currently, the product displays all the TAs at once.

To facilitate easy of navigation for users, default course is set up using the TeachCommand.

This is so that the next time users log into the product, they can directly see the list of TAs teaching under them.

Closes #45.

### CODE REVIEW

File: src/main/java/seedu/address/model/Model.java [hunk 1]
```diff
@@ -44,6 +45,17 @@ public interface Model {
...
+     * Returns the module the user is teaching.
+     *
+     */
+    Mod getTeaching();
+
+    /**
+     * Sets the module the user is teaching.
+     */
+    void setTeaching(Mod module);
+
```
Comment (lshaoqin) [inline src/main/java/seedu/address/model/Model.java]: I don't think these methods should be part of the Model interface, as they might not be implemented in other models if we choose to extend it.

File: src/main/java/seedu/address/model/UserPrefs.java [hunk 2]
```diff
@@ -16,6 +17,7 @@ public class UserPrefs implements ReadOnlyUserPrefs {
     private GuiSettings guiSettings = new GuiSettings();
     private Path addressBookFilePath = Paths.get("data" , "addressbook.json");
 
+    private Mod teaching;
```
Comment (lshaoqin) [inline src/main/java/seedu/address/model/UserPrefs.java]: How does this field influence startup behavior at the moment?

**Review by lshaoqin — COMMENTED** [review]
I need some clarification as to what the intended functionality of the comment is.

Code is clean overall but testcases are needed. In addition, the refactoring of Mod into Course would affect this PR greatly. Let's wait for #40 to be resolved before continuing work on this.

File: src/main/java/seedu/address/model/Model.java (hunk 1 above)
Comment (itssisi) [reply src/main/java/seedu/address/model/Model.java]: It is necessary for the `Logic` class to fetch the default course. So, it is still there.

File: src/main/java/seedu/address/model/UserPrefs.java (hunk 2 above)
Comment (itssisi) [reply src/main/java/seedu/address/model/UserPrefs.java]: The title of the window changes (refer to screenshot). The list of TAs are filtered according to the default course automatically.

File: src/main/java/seedu/address/logic/commands/TeachCommand.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+/**
+ * Adds the default course for users in the address book.
+ */
+public class TeachCommand extends Command {
+    public static final String COMMAND_WORD = "teach";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sets up default course users are teaching.\n"
+            + "Parameters: "
+            + PREFIX_COURSE + "COURSE \n"
```
Comment (lshaoqin) [inline src/main/java/seedu/address/logic/commands/TeachCommand.java]: let's standardise this as COURSE_CODE

**Review by lshaoqin — CHANGES_REQUESTED** [review]
Good job! Nice reuse of TeachingCoursePredicate here.

Two bugs that I found:
1) After setting the course to teach, the find command can still find TAs from outside the course. Not sure if this is intended behaviour.
2) `find n/` will result in `java.lang.reflect.InvocationTargetException` but no error message is shown to the user in the app. Tagging @raynertjx for this.

Also, perhaps there should be a way to reset the addressbook to having no default course?

File: src/main/java/seedu/address/logic/commands/TeachCommand.java (hunk 3 above)
Comment (itssisi) [reply src/main/java/seedu/address/logic/commands/TeachCommand.java]: Bug number 1 is related to @raynertjx. 

In our current feature, once the default course is set up, it is not meant to be reset from the product. We can do it manually by deleting the default course at the json file.

Comment (itssisi) [reply src/main/java/seedu/address/logic/commands/TeachCommand.java]: > let's standardise this as COURSE_CODE

fixed

**Review by lshaoqin — APPROVED** [review]

**Pull request merged**